File: cst_to_utc/cst_to_utc_nim/cst_to_utc.nim
import std/os
import std/strutils

# echo paramCount(), " ", paramStr(1)

# Check number of arguments
if paramCount() != 2:
  echo "Invalid arguments!"
  echo "Usage: ", paramStr(0), " <hour> <AM or PM>"
  quit()

# Set variables from arguments
var hour: int = parseInt(paramStr(1))
var day_night: string = paramStr(2)

# Check if arguments are correct
if hour > 23:
  echo "Invalid arguments! \"hour\" does not exist"
  echo "Usage: ", paramStr(0), " <hour> <AM or PM>"
  quit()

if day_night != "AM" and day_night != "PM":
  echo "Invalid arguments! Please use AM or PM"
  echo "Usage: ", paramStr(0), " <hour> <AM or PM>"
  quit()

# Convert time
if day_night == "PM":
  var new_hour: int = hour + 12
  var utc: int = new_hour + 6
  if utc > 23:
    echo utc - 24, " AM UTC the next day"
  else:
    echo utc, " PM UTC"
else:
  var utc: int = hour + 6
  if utc >= 12:
    echo utc, " PM UTC"
  else:
    echo utc, " AM UTC"

File: cst_to_utc/cstToUtc.py
#!/usr/bin/python3

import sys

# Check inputs
if len(sys.argv) != 3:
    print("Missing arguments!")
    print("Usage: cstToUtc.py <hour> <AM or PM>")
    exit()

# Set variables from inputs
hour = int(sys.argv[1])
dayNight = str(sys.argv[2])

# Check if arguments are correct
if not isinstance(hour, int) or hour > 23:
    print("Invalid arguments! \"Hour\" does not exist.")
    print("Usage: cstToUtc.py <hour> <AM or PM>")
    exit()

if dayNight != "PM" and dayNight != "AM":
    print("Invalid arguments! Please use AM or PM.")
    print("Usage: cstToUtc.py <hour> <Am or PM>")
    exit()

# Convert time 
if dayNight == str("PM"):
    NewHour = int(hour) + 12
    UTC = NewHour + 6
    if UTC > 23:
        print(f"{UTC - 24} AM UTC the next day")
    else:
        print(f"{UTC} PM UTC")
else:
    UTC = int(hour) + 6
    if UTC >= 12:
        print(f"{UTC} PM UTC")
    else:
        print(f"{UTC} AM UTC")
